example/README.md

```diff
@@ -0,0 +1 @@
+This is a simple live demo of `restul-react` components primarily used to test features. Run it with `npm run example`.
```

Comment:
We already have codesandbox example and unit tests. If you can avoid to introduce something else to maintain… And I'm sorry but this is not replacing unit tests. I can help you to add the correct unit tests for this feature if needed but please remove this.

Reply:
This is just a `dev-server` kind of like in Operational. It doesn't really add much bloat to the library. It's just an on-device, in-browser player like GraphiQL. Ideally, we don't maintain this but make sure our features work _in a browser context_.

I think the tests:
- should be maintained
- add a _lot_ of value

_but_, there is no guarantee that they work 100% in a browser environment with calls to _real_ APIs. This will serve as a check we do _during development_ to make sure things work before pushing.

I don't really see any major costs to having this around. Do you?

Reply:
@peterszerzo he does have a point though – we _should_ ideally add unit tests for the `resolve` behavior as well.

Reply:
I see this more as an integration test as opposed to a unit test - it makes sure that a real request can be made, goes through the state management, and renders something in the end. Which is something I am not confident a unit test suite would catch, especially after a feature this complicated.

Reply:
I don't dispute the need for improving the tests for the new `resolve` - I just don't feel comfortable shipping this feature without validating it with an end-to-end example. This could even grow to be like the visual test cases suite in https://github.com/contiamo/operational-visualizations

Reply:
I got your point, for now I just see this dev server as an escape to implement real tests… After why not, but if we go with an example folder, I will prefer something that we can deploy on codesandbox with more examples as a real addition to our documentation.

I'm just worry about the "it's works on my little example, let's skip the tests for now" 😉

Reply:
And at this library stage, I'm not agree with you, **we must have** unit tests for every features. Without tests, we can't be sure that the functionally works as expected, it's hard to maintained and everybody can broke the feature by mistake (and by everybody, I include myself of course).

Reply:
I'm maybe a bit strict on this stack, but all our product depends on this library :wink:

Reply:
> I'm just worry about the "it's works on my little example, let's skip the tests for now" 😉

As long as we have you on the team, I am confident that this will never happen. 😉  Let's also look at making the examples richer and exportable to CodeSandbox (thanks, by the way, @CompuIves) at a later stage.

Reply:
@fabien0102 we don't have to merge until we have a satisfactory test suite, and that would have been the case whether we had the example server or not. Both methods of testing are good for reliability and for catching issues that can bubble up in dependent projects - there is no reason to jump against one of them.

Reply:
@peterszerzo I agree that both methods of testing are good in the context of a PR, but our goal is to have a maintainable product in the long term. For this purpose, unit tests ensure a non regression in the future, it's automated in the CI, etc etc... So no, both methods of testing are not equal in this case.